vulnerabilities/tests/test_data_source.py

```diff
@@ -7,330 +7,34 @@
-                subdir="crates", recursive=True, file_ext="toml"
-            )
-
-        assert len(added_files) >= 2
-        assert os.path.join(self.repodir, "crates/bitvec/RUSTSEC-2020-0007.toml") in added_files
-        assert os.path.join(self.repodir, "crates/hyper/RUSTSEC-2020-0008.toml") in added_files
-        assert len(updated_files) == 0
-
-    def test_file_changes_include_fixed_advisories(self):
-        # pick a date that includes commit 9889ed0831b4fb4beb7675de361926d2e9a99c20
-        # ("Fix patched version for RUSTSEC-2020-0008")
-        last_run_date = datetime.datetime(
-            year=2020, month=3, day=31, hour=17, minute=40, tzinfo=datetime.timezone.utc
-        )
-        ds = self.mk_ds(last_run_date=last_run_date, cutoff_date=None)
-
-        with ds:
-            added_files, updated_files = ds.file_changes(
-                subdir="crates", recursive=True, file_ext="toml"
-            )
-
-        assert len(added_files) == 0
-        assert len(updated_files) == 1
-        assert os.path.join(self.repodir, "crates/hyper/RUSTSEC-2020-0008.toml") in updated_files
-
-
-class TestOvalImporter(TestCase):
-    @classmethod
-    def setUpClass(cls):
-        cls.oval_data_src = OvalImporter(1)
-
-    def test_create_purl(self):
-        purl1 = PackageURL(name="ffmpeg", type="test", version="1.2.0")
-
-        assert purl1 == self.oval_data_src.create_purl(
-            pkg_name="ffmpeg", pkg_version="1.2.0", pkg_data={"type": "test"}
-        )
-
-        purl2 = PackageURL(
-            name="notepad",
-            type="example",
-            version="7.9.6",
-            namespace="ns",
-            qualifiers={"distro": "sample"},
-            subpath="root",
-        )
-        assert purl2 == self.oval_data_src.create_purl(
-            pkg_name="notepad",
-            pkg_version="7.9.6",
-            pkg_data={
-                "namespace": "ns",
-                "qualifiers": {"distro": "sample"},
-                "subpath": "root",
-                "type": "example",
-            },
-        )
-
-    def test__collect_pkgs(self):
-
-        xmls = load_oval_data()
-
-        expected_suse_pkgs = {"cacti-spine", "apache2-mod_perl", "cacti", "apache2-mod_perl-devel"}
-        expected_ubuntu_pkgs = {"potrace", "tor"}
-
-        translations = {"less than": "<"}
-
-        found_suse_pkgs = self.oval_data_src._collect_pkgs(
-            OvalParser(translations, xmls["suse"]).get_data()
-        )
-
-        found_ubuntu_pkgs = self.oval_data_src._collect_pkgs(
-            OvalParser(translations, xmls["ubuntu"]).get_data()
-        )
-
-        assert found_suse_pkgs == expected_suse_pkgs
-        assert found_ubuntu_pkgs == expected_ubuntu_pkgs
+class TestGitImporter(TestCase):
+    def test_clone_valid(self):
+        with patch.object(GitImporter, "__init__", return_value=None):
+            c = GitImporter(None)
+            c.repo_url = "git+https://github.com/nexB/fetchcode"
```

Comment:
Does this do a live clone over the network?

Reply:
yes, it clones the repository
( I just mock a constructor and then make the Git importer run, so I can test the result )